### PR TITLE
Add ascii() and support for multibyte Unicode

### LINKED ISSUE #812: Document ascii() and the support for multibyte unicode for overlay and ascii functions
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/9321

### Which part(s) of the docs might be affected or should be updated? And how?

SQL - Functions - String 

### Reference

Go to e2e_test-> batch -> functions , and also the PG documentation for reference.

### PR DESCRIPTION
**Pull request by bernscode**
Add ascii() and support for multibyte Unicode characters for ascii() and overlay().

## Info
- **Description**: 
Add ascii() and support for multibyte Unicode characters for ascii() and overlay().

- **Preview**: 
https://pr-849.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-function-string/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/9321

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/812

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

**Review by xiangjinwu — COMMENTED** [review]
All string functions are supposed to support multibyte Unicode characters. The PR just fixes the ones that fail to support them initially. We do not need to emphasize it per function. Maybe just add some examples to demonstrate the support?
`ascii('🌊')` → `127754`

**Comment by bernscode** [comment]
@xiangjinwu Good point! I've made the update to the latest commit. Thanks for the feedback!

File: docs/sql/functions-operators/sql-function-string.md [hunk 1]
```diff
@@ -30,7 +31,7 @@ title: String functions and operators
...
-|overlay ( *input_string* PLACING *substring* FROM *start_int* [ FOR *length_int* ]) → *output_string* | Replaces a substring in *input_string* with *substring*, starting at *start_int* and with *length_int*. If *length_int* is omitted, its value is the length of *substring*.| `overlay('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `overlay('abcdef' PLACING '45' FROM 4)` → `abc45f`|
+|overlay ( *input_string* PLACING *substring* FROM *start_int* [ FOR *length_int* ]) → *output_string* | Replaces a substring in *input_string* with *substring*, starting at *start_int* and with *length_int*. If *length_int* is omitted, its value is the length of *substring*. | `overlay('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `overlay('abcdef' PLACING '45' FROM 4)` → `abc45f` <br /> `overlay('RisingWave' PLACING '🌊' FROM 7)` → `Rising🌊Wave` |
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-string.md]: `overlay('RisingWave' PLACING '🌊' FROM 7)` → `Rising🌊ave`

where the 🌊 is overlayed on the 7th character W

**Review by xiangjinwu — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-string.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-string.md]: @xiangjinwu Thanks for catching that typo. I've made the update.

**Review by hengm3467 — APPROVED** [review]
LGTM

**Comment by xiangjinwu** [comment]
Seems the `replace` function was lost during merge main

**Review by CharlieSYH — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-string.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ title: String functions and operators
...
-|`overlay` ( *input_string* PLACING *substring* FROM *start_int* [ FOR *length_int* ]) → *output_string* | Replaces a substring in *input_string* with *substring*, starting at *start_int* and with *length_int*. If *length_int* is omitted, its value is the length of *substring*.| `overlay('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `overlay('abcdef' PLACING '45' FROM 4)` → `abc45f`|
+|overlay ( *input_string* PLACING *substring* FROM *start_int* [ FOR *length_int* ]) → *output_string* | Replaces a substring in *input_string* with *substring*, starting at *start_int* and with *length_int*. If *length_int* is omitted, its value is the length of *substring*. | `overlay('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `overlay('abcdef' PLACING '45' FROM 4)` → `abc45f` <br /> `overlay('RisingWave' PLACING '🌊' FROM 7)` → `Rising🌊ave` |
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-string.md]:
```suggestion
|`overlay` ( *input_string* PLACING *substring* FROM *start_int* [ FOR *length_int* ]) → *output_string* | Replaces a substring in *input_string* with *substring*, starting at *start_int* and with *length_int*. If *length_int* is omitted, its value is the length of *substring*. | `overlay('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `overlay('abcdef' PLACING '45' FROM 4)` → `abc45f` <br /> `overlay('RisingWave' PLACING '🌊' FROM 7)` → `Rising🌊ave` |
```

File: docs/sql/functions-operators/sql-function-string.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ title: String functions and operators
 
 |Function|Description|Example|
 |---|---|---|
-|`replace` ( *input_string*, *from_string*, *to_string* ) → *output_string*|Replaces all occurrences of substring *from_string* in *input_string* with substring *to_string*.|`replace('abcdefabcdef', 'cd', 'XX')` → `abXXefabXXef`|
+| ascii( *input_string* ) → *int* | Returns the Unicode code point of the first character of the *input_string*. If the string is empty, it returns `NULL`. | `ascii('RisingWave')` → `82` <br /> `ascii('🌊')` → `127754` |
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-string.md]:
```suggestion
| `ascii`( *input_string* ) → *int* | Returns the Unicode code point of the first character of the *input_string*. If the string is empty, it returns `NULL`. | `ascii('RisingWave')` → `82` <br /> `ascii('🌊')` → `127754` |
```

Comment (CharlieSYH) [reply docs/sql/functions-operators/sql-function-string.md]: I tested using the code format for function names on this page. RN the function names are hard to identify when you browse through function pages. Let's add the `` for function names on this page.

File: docs/sql/functions-operators/sql-function-string.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ title: String functions and operators
 
 |Function|Description|Example|
 |---|---|---|
-|`replace` ( *input_string*, *from_string*, *to_string* ) → *output_string*|Replaces all occurrences of substring *from_string* in *input_string* with substring *to_string*.|`replace('abcdefabcdef', 'cd', 'XX')` → `abXXefabXXef`|
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-string.md]: Please restore the `replace` function. It should not be deleted.

**Review by CharlieSYH — CHANGES_REQUESTED** [review]

File: docs/sql/functions-operators/sql-function-string.md (hunk 4 above)
Comment (bernscode) [reply docs/sql/functions-operators/sql-function-string.md]: @CharlieSYH Thanks for pointing this out. I've re-added the `replace` function on the latest commit.

**Review by CharlieSYH — APPROVED** [review]

**Pull request merged**